samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -247,7 +247,6 @@ StreamProcessor createStreamProcessor(
     latch.await();
 
     if (throwable.get() != null) {
-      appStatus = ApplicationStatus.UnsuccessfulFinish;
       throw throwable.get();
```

Comment:
oh, I removed this line is because it's redundant. As you see, the exception thrown here will be caught in line 136 and the status will be set again. The code is pretty messy right now. I am waiting for @navina 's change to finally clean these up in patch (https://github.com/apache/samza/pull/135). I will also add the waitForFinish() in that patch.